PC/getpathp.c

```diff
@@ -265,7 +264,8 @@ canonicalize(wchar_t *buffer, const wchar_t *path)
         return _PyStatus_NO_MEMORY();
     }
 
-    if (PathIsRelativeW(path)) {
+    const wchar_t *pathTail;
+    if (FAILED(PathCchSkipRoot(path, &pathTail)) || path == pathTail) {
```

Comment:
Are you not worried at all that `canonicalize()` will ever be called with paths that contain forward slashes? All of the `PathCch*` functions handle forward slash as just a name character, in which case skipping the root, combining with current directory, and getting a canonical path all misbehave badly. It wouldn't hurt to add a `wcschr()` call to check for forward slashes. Though another large buffer would have to be allocated on the stack in case `path` has to be copied to modify it.

Reply:
Nah, it's shipped already and we'd have heard about issues if they exist. Plus #29041 is going to delete this code and replace it with code that handles both anyway.